src/plugins/link/main/ts/api/Settings.ts

```diff
@@ -5,10 +5,22 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
+import { ForceExternalTargets } from './Types';
+
 const assumeExternalTargets = function (editorSettings) {
   return typeof editorSettings.link_assume_external_targets === 'boolean' ? editorSettings.link_assume_external_targets : false;
 };
 
+const forceExternalTargets = function (editorSettings) {
```

Comment:
Does the enum actually help here, instead of just a union type given it's user input? As it seems to just be causing more if statements than needed.

So you could just remove the enum entirely and make the setting return type be `'https' | 'http' | undefined`. That way you can remove all the if logic in the settings:

```
const forceExternalTargets = function (editorSettings): 'https' | 'http' | undefined {
  const external_targets = editorSettings.link_force_external_targets;
  return external_targets === 'http' || external_targets === 'https' ? external_targets : undefined;
};
```

and instead just do something like this in the utils, since you can use the types to validate at this point:

```
const handleExternalTargets = (href: string, forceExternalTargets: 'https' | 'http' | undefined): string => {
  if (forceExternalTargets !== undefined && !protocolRe.test(href)) {
    return forceExternalTargets + '://' + href;
  }
  return href;
};
```

Another option, which is probably the better of the two, would be to give the enums a corresponding value so you can use them directly, instead of needing the additional if statements. eg:

```
export const enum ForceExternalTargets {
  NO,
  ALWAYS_HTTP = 'http',
  ALWAYS_HTTPS = 'https'
}
```
and
```
const handleExternalTargets = (href: string, forceExternalTargets: ForceExternalTargets): string => {
  if (forceExternalTargets !== ForceExternalTargets.NO && !protocolRe.test(href)) {
    return forceExternalTargets + '://' + href;
  }
  return href;
};
```